lib/orangelight/middleware/invalid_parameter_handler.rb

```diff
@@ -46,24 +50,43 @@ def default_content_type
           'text/html'
         end
 
-        def request_content_type
-          @request.formats.first || default_content_type
+        # Check if facet fields have empty value lists
+        def facet_fields_values(params)
+          params.fetch(:f, []).collect do |facet_field, value_list|
+            next unless value_list.nil?
+            raise ActionController::BadRequest, "Facet field #{facet_field} has a nil value"
+          end
         end
 
-        def bad_request_headers
-          {
-            'Content-Type' => "#{request_content_type}; charset=#{default_charset}",
-            'Content-Length' => bad_request_body.bytesize.to_s
-          }
+        def raise_error?(message)
+          valid_message_patterns.each do |pattern|
+            return false if message.match?(pattern)
+          end
         end
 
-        def bad_request_response
+        def request_content_type(env)
+          request = request_for(env)
+          request.formats.first || default_content_type
+        end
+
+        def request_for(env)
+          # calling env.dup here prevents bad things from happening
+          ActionDispatch::Request.new(env.dup)
```

Comment:
Caching the request was causing problems because that cached value persists between calls. Unfortunately, this adds some complexity to the methods because we have to pass the `env` variable around more than before.